packages/parser/tests/unit/envelopes/dynamodb.test.ts

```diff
@@ -1,141 +1,206 @@
-import { generateMock } from '@anatine/zod-mock';
-import type { AttributeValue, DynamoDBStreamEvent } from 'aws-lambda';
+import { marshall } from '@aws-sdk/util-dynamodb';
 import { describe, expect, it } from 'vitest';
 import { ZodError, z } from 'zod';
 import { DynamoDBStreamEnvelope } from '../../../src/envelopes/index.js';
 import { ParseError } from '../../../src/errors.js';
-import { TestEvents } from '../schema/utils.js';
+import type { DynamoDBStreamEvent } from '../../../src/types/schema.js';
+import { getTestEvent } from '../schema/utils.js';
 
-describe('DynamoDB', () => {
-  const schema = z.object({
-    Message: z.record(z.literal('S'), z.string()),
-    Id: z.record(z.literal('N'), z.number().min(0).max(100)),
+describe('Envelope: DynamoDB Stream', () => {
+  const schema = z
+    .object({
+      Message: z.string(),
+      Id: z.number(),
+    })
+    .strict();
+  const baseEvent = getTestEvent<DynamoDBStreamEvent>({
+    eventsPath: 'dynamodb',
+    filename: 'base',
   });
-  const mockOldImage = generateMock(schema);
-  const mockNewImage = generateMock(schema);
-  const dynamodbEvent = TestEvents.dynamoStreamEvent as DynamoDBStreamEvent;
-  // biome-ignore lint/style/noNonNullAssertion: it is ensured that this event has these properties
-  (dynamodbEvent.Records[0].dynamodb!.NewImage as typeof mockNewImage) =
-    mockNewImage;
-  // biome-ignore lint/style/noNonNullAssertion: it is ensured that this event has these properties
-  (dynamodbEvent.Records[1].dynamodb!.NewImage as typeof mockNewImage) =
-    mockNewImage;
-  // biome-ignore lint/style/noNonNullAssertion: it is ensured that this event has these properties
-  (dynamodbEvent.Records[0].dynamodb!.OldImage as typeof mockOldImage) =
-    mockOldImage;
-  // biome-ignore lint/style/noNonNullAssertion: it is ensured that this event has these properties
-  (dynamodbEvent.Records[1].dynamodb!.OldImage as typeof mockOldImage) =
-    mockOldImage;
-  describe('parse', () => {
-    it('parse should parse dynamodb envelope', () => {
-      const parsed = DynamoDBStreamEnvelope.parse(dynamodbEvent, schema);
-      expect(parsed[0]).toEqual({
-        OldImage: mockOldImage,
-        NewImage: mockNewImage,
-      });
-      expect(parsed[1]).toEqual({
-        OldImage: mockOldImage,
-        NewImage: mockNewImage,
-      });
-    });
-    it('parse should throw error if envelope invalid', () => {
+
+  describe('Method: parse', () => {
+    it('throws if one of the payloads does not match the schema', () => {
+      // Prepare
+      const event = structuredClone(baseEvent);
+
+      // Act & Assess
       expect(() =>
-        DynamoDBStreamEnvelope.parse({ foo: 'bar' }, schema)
-      ).toThrow();
+        DynamoDBStreamEnvelope.parse(
+          event,
+          z
+            .object({
+              Message: z.string(),
+            })
+            .strict()
+        )
```

Comment:
Is that default biome formatting?

Reply:
Yep

Reply:
I tried again to put it better, and it put it back when formatting.